libs/common/src/lib/search-widget/widgets/widget-generator.component.ts

```diff
@@ -669,13 +674,22 @@ export class WidgetGeneratorComponent implements OnInit, OnDestroy {
   generativemodel="${this.generativeModel}"`
         : '';
       const mode: string = this.darkModeEnabled ? `mode="dark"` : '';
-      const baseSnippet = `<nuclia-search-bar ${mode}
+      const tagName = this.popupModeEnabled ? 'nuclia-popup' : 'nuclia-search-bar';
+      let baseSnippet = `<${tagName} ${mode}
   knowledgebox="${kb.id}"
   ${zone}
-  features="${this.features}" ${ragProperties}${ragImagesProperties}${placeholder}${notEnoughDataMessage}${askToResource}${generativeModel}${filters}${preselectedFilters}${privateDetails}${backend}></nuclia-search-bar>
+  features="${this.features}" ${ragProperties}${ragImagesProperties}${placeholder}${notEnoughDataMessage}${askToResource}${generativeModel}${filters}${preselectedFilters}${privateDetails}${backend}></${tagName}>`;
+      if (!this.popupModeEnabled) {
+        baseSnippet += `
 <nuclia-search-results ${mode}></nuclia-search-results>`;
+      } else {
+        baseSnippet += `
+<div data-nuclia="search-widget-button">Click here to open the Nuclia search widget</div>`;
```

Comment:
why did you make a div instead of a button or a link (something semantically clickable)?